tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaCommon.cs

```diff
@@ -32,8 +32,17 @@ internal static Scope CreatePlaceholderScope(Tracer tracer, string traceId, stri
         }
         else
         {
-            Log($"creating the placeholder traceId = {traceId}");
-            span = tracer.StartSpan(PlaceholderOperationName, tags: null, serviceName: PlaceholderServiceName, traceId: (TraceId)Convert.ToUInt64(traceId), addToTraceContext: false);
+            if (traceIdUpper64 == 0)
+            {
+                Log($"creating the placeholder traceId = {traceId}");
+                span = tracer.StartSpan(PlaceholderOperationName, tags: null, serviceName: PlaceholderServiceName, traceId: (TraceId)Convert.ToUInt64(traceId), addToTraceContext: false);
+            }
+            else
+            {
+                var traceIdLower64 = Convert.ToUInt64(traceId);
+                Log($"creating the placeholder traceId = {traceIdUpper64}{traceIdLower64}");
+                span = tracer.StartSpan(PlaceholderOperationName, tags: null, serviceName: PlaceholderServiceName, traceId: new TraceId(traceIdUpper64, traceIdLower64), addToTraceContext: false);
+            }
```

Comment:
You can pass `traceIdUpper64` directly to the `TraceId` constructor without the `if` to handle the two cases:

```suggestion
            var traceId = new TraceId(traceIdUpper64, traceIdLower64);
            Log($"creating the placeholder traceId = {traceId}");
            span = tracer.StartSpan(PlaceholderOperationName, tags: null, serviceName: PlaceholderServiceName, traceId: traceId, addToTraceContext: false);
```

(this also fixes the trace id in the log I mentioned in the other comment above)